page.go

```diff
@@ -288,29 +320,37 @@ func renderPackagePage(resp http.ResponseWriter, req *http.Request, repo *Repo)
 				gopkgPath := repo.GopkgPath()
 				for _, result := range searchResults.Results {
 					if result.Path == gopkgPath {
-						dataMutex.Lock()
-						data.Synopsis = result.Synopsis
-						dataMutex.Unlock()
+						synopsis <- result.Synopsis
 						break
 					}
 				}
 			}
 		}
-		gotResp <- true
 	}()
 
+	wantResps := 2
```

Comment:
You could do this below as part of the for loop:

for r, wantResps := 0, 2; r < wantResps; {